bridges/prometheus/producer.go

```diff
@@ -0,0 +1,254 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus // import "go.opentelemetry.io/contrib/bridges/prometheus"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+const (
+	scopeName    = "go.opentelemetry.io/otel/bridge/prometheus"
+	traceIDLabel = "trace_id"
+	spanIDLabel  = "span_id"
+)
+
+var (
+	errUnsupportedType = errors.New("unsupported metric type")
+	processStartTime   = time.Now()
+)
+
+type producer struct {
+	gatherers []prometheus.Gatherer
```

Comment:
Isn't this type the same as [`prometheus.Gatherers`](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Gatherers)?  is there a reason this isn't used?

Reply:
updated